tests/unit/_modules/test_packagemanager.py

```diff
@@ -0,0 +1,345 @@
+import pytest
+from srv.salt._modules.packagemanager import PackageManager, Zypper, Apt
+from mock import MagicMock, patch, mock
+
+
+class TestPackageManager():
+    '''
+    This class contains a set of functions that test srv.salt._modules.packagemanager
+    '''
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_opensuse(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('opensuse', 42.1, 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Zypper) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_suse(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('SUSE', '12.2', 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Zypper) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_centos(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('centos', '8', 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Apt) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_fedora(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('fedora', '23', 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Apt) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    @mock.patch('srv.salt._modules.packagemanager.Popen')
+    def test_reboot(self, po, dist_return):
+        """
+        Reboot method
+        """
+        dist_return.return_value = ('opensuse', '42.2', 'x86_64')
+        ret = PackageManager()
+        ret.reboot_in()
+        assert po.called is True
```

Comment:
If I'm reading this correctly, this is only asserting that Popen is called, but it doesn't specify what Popen should be doing. For a reboot, we want to assert that Popen is called and that the command issued is a reboot, otherwise this test isn't very explicit.

I see this elsewhere in this test module too.